### PR TITLE
Make CoreBundle optional

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this changes are BC.

Part of https://github.com/sonata-project/dev-kit/issues/697

## Changelog

```markdown
### Removed
- remove SonataCoreBundle dependencies
```

### CODE REVIEW

File: src/SonataCommentBundle.php [hunk 1]
```diff
@@ -51,16 +44,18 @@ public function boot()
...
-            'fos_comment_comment' => 'FOS\CommentBundle\Form\CommentType',
-            'fos_comment_commentable_thread' => 'FOS\CommentBundle\Form\CommentableThreadType',
-            'fos_comment_delete_comment' => 'FOS\CommentBundle\Form\DeleteCommentType',
-            'fos_comment_thread' => 'FOS\CommentBundle\Form\ThreadType',
-            'fos_comment_vote' => 'FOS\CommentBundle\Form\VoteType',
-            'sonata_comment_comment' => 'Sonata\CommentBundle\Form\Type\CommentType',
-            'sonata_comment_status' => 'Sonata\CommentBundle\Form\Type\CommentStatusType',
-        ]);
+        if (class_exists('Sonata\CoreBundle\Form\FormHelper')) {
+            \Sonata\CoreBundle\Form\FormHelper::registerFormTypeMapping([
```
Comment (greg0ire) [inline src/SonataCommentBundle.php:50]: You can still have it in a use statement, please revert that change.

Comment (wbloszyk) [reply src/SonataCommentBundle.php:50]: Can i change this methods to private? It is potentialy BC-break but i can't imagine case when someone using it or extending.

Comment (greg0ire) [reply src/SonataCommentBundle.php:52]: It's pretty unlikely, but do you _need_ to?

**Comment by jordisala1991** [comment]
Thank you @wbloszyk

**Pull request merged**